src/saturn_engine/utils/error_handling.py

```diff
@@ -0,0 +1,148 @@
+    else:
+        module, _, klass = exception_type.rpartition(".")
+        m = __import__(module, None, None, [klass])
+    return getattr(m, klass)
+
+
+def does_output_match(
+    exception: ExceptionDetails, exception_filter: ExceptionFilter
+) -> bool:
+    return (
+        (
+            exception_filter.message == ""
+            or re.match(exception_filter.message, exception.message) is not None
+        )
+        and issubclass(exception.exception_type, exception_filter.exception_type)
+        and (
+            exception_filter.module == ""
+            or exception.module.endswith(exception_filter.module)
+        )
+        and (
+            exception_filter.lineno == 0 or exception.lineno == exception_filter.lineno
+        )
+    )
+    return a and b and c and d
+
+
+def file_to_module(path: str) -> str:
+    basefile, _ = os.path.splitext(path)
+    return basefile.replace(os.path.sep, ".")
+
+
+def get_exception_details() -> ExceptionDetails:
+    exc_type, exc_value, exc_traceback = sys.exc_info()
+    if exc_type is RemoteException:
+        if exc_value.remote_traceback.exc_module == "builtins":
+            exc_type = resolve_exception(exc_value.remote_traceback.exc_type)
+        else:
+            exc_type = ".".join(
+                [
+                    exc_value.remote_traceback.exc_module,
+                    exc_value.remote_traceback.exc_type,
+                ]
+            )
+            exc_type = resolve_exception(exc_type)
+
+    return ExceptionDetails(
+        message=str(exc_value),
+        exception_type=exc_type,
+        module=file_to_module(exc_traceback.tb_frame.f_code.co_filename),
+        lineno=exc_traceback.tb_lineno,
+    )
+
+
+def is_output_error_handler(output: str) -> bool:
+    return output.startswith("error:") or output.startswith("ignore:")
+
+
+@lru_cache(maxsize=50)
+def parse_warning_filter(arg: str, *, escape: bool) -> ExceptionFilter:
+    """Parse a warnings filter string.
+    This is copied from warnings._setoption with the following changes:
+    * Does not apply the filter.
+    * Escaping is optional.
+    """
+    __tracebackhide__ = True
+    error_template = dedent(
+        f"""\
+        while parsing the following warning configuration:
+          {arg}
+        This error occurred:
+        {{error}}
+        """
+    )
+
+    parts = arg.split(":")
+    if len(parts) > 5:
+        doc_url = (
+            "https://docs.python.org/3/library/warnings.html#describing-warning-filters"
+        )
+        error = dedent(
+            f"""\
+            Too many fields ({len(parts)}), expected at most 5 separated by colons:
+              action:message:category:module:line
+            For more information please consult: {doc_url}
+            """
+        )
+        raise Exception(error_template.format(error=error))
+
+    while len(parts) < 5:
+        parts.append("")
+
+    action, message, category, module, lineno_ = (s.strip() for s in parts)
```

Comment:
`action` should not be used. Here `error:` in out channel name means this channel handle error, it doesn't mean how these error are handled. The topic are what's use for know how they are being handled.